scalardb-test/src/main/java/kelpie/scalardb/transfer/graphql/HttpTransferProcessorWithRetry.java

```diff
@@ -0,0 +1,255 @@
+          + "  ) {\n"
+          + "    tx_transfer {\n"
+          + "      balance\n"
+          + "    }\n"
+          + "  }\n"
+          + "}";
+
+  private static final String PUT_BALANCES =
+      "mutation updateBalances($tx_id: String!,\n"
+          + "                  $from_id: Int!, $from_type: Int!, $from_new_balance: Int,\n"
+          + "                  $to_id: Int!, $to_type: Int!, $to_new_balance: Int)"
+          + "  @transaction(id: $tx_id, commit: true) {\n"
+          + "  from_put: tx_transfer_put(\n"
+          + "    put: {key: {account_id: $from_id, account_type: $from_type},\n"
+          + "          values: {balance: $from_new_balance}}\n"
+          + "  )\n"
+          + "  to_put: tx_transfer_put(\n"
+          + "    put: {key: {account_id: $to_id, account_type: $to_type},\n"
+          + "          values: {balance: $to_new_balance}}\n"
+          + "  )\n"
+          + "}";
+
+  private final int numAccounts;
+  private final String endpointUrl;
+  private final OkHttpClient client;
+  private final ThreadLocalCookieJar threadLocalCookieJar;
+  private final Retry retry;
+  private final LongAdder retryCount = new LongAdder();
+
+  public HttpTransferProcessorWithRetry(Config config) {
+    super(config);
+
+    this.numAccounts = (int) config.getUserLong("test_config", "num_accounts");
+    endpointUrl = config.getUserString("graphql_http", "endpoint_url");
+
+    threadLocalCookieJar = new ThreadLocalCookieJar();
+
+    int concurrency = (int) config.getUserLong("common", "concurrency");
+    client =
+        new OkHttpClient.Builder()
+            .readTimeout(Duration.ofSeconds(60))
+            .connectTimeout(Duration.ofSeconds(60))
+            .connectionPool(new ConnectionPool(concurrency, 5, TimeUnit.SECONDS))
+            .cookieJar(threadLocalCookieJar)
+            .build();
+
+    RetryConfig retryConfig =
+        RetryConfig.custom()
+            .maxAttempts(MAX_RETRIES)
+            .intervalFunction(
+                IntervalFunction.ofExponentialBackoff(INITIAL_INTERNAL_MILLS, MULTIPLIER))
+            .retryOnException(
+                t ->
+                    t instanceof GraphQlFailureException
+                        && ((GraphQlFailureException) t).isRetryable())
+            .build();
+    retry = Retry.of("HttpTransferProcessorWithRetry", retryConfig);
+    retry
+        .getEventPublisher()
+        .onRetry(
+            e -> {
+              logWarn(e.toString(), e.getLastThrowable());
+              retryCount.increment();
+            });
+  }
+
+  @Override
+  public void executeEach() throws Exception {
+    int fromId = ThreadLocalRandom.current().nextInt(numAccounts);
+    int toId = ThreadLocalRandom.current().nextInt(numAccounts);
+    int amount = ThreadLocalRandom.current().nextInt(1000) + 1;
+
+    try {
+      Retry.decorateCheckedRunnable(retry, () -> transfer(fromId, toId, amount)).run();
```

Comment:
[minor] How about inheriting `HttpTransferProcessor` and calling the parent's `transfer` so that we don't need to maintain the duplicated code? Of course, using inheritance in every case isn't good, but I think it's okay in this case.

Reply:
Yeah, that sounds good. I think we can apply the idea for other cases in this project. Let me do that in another PR.